components/ProposalVotingPower/LockedCommunityNFTRecordVotingPower.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useState, useEffect } from 'react'
+import classNames from 'classnames'
+import { BigNumber } from 'bignumber.js'
+import useRealm from '@hooks/useRealm'
+import useProposal from '@hooks/useProposal'
+import useWalletStore from 'stores/useWalletStore'
+import useHeliumVsrStore from 'HeliumVotePlugin/hooks/useHeliumVsrStore'
+import { fmtMintAmount, getMintDecimalAmount } from '@tools/sdk/units'
+import { getMintMetadata } from '@components/instructions/programs/splToken'
+import Tooltip from '@components/Tooltip'
+import { ChevronRightIcon, LightningBoltIcon } from '@heroicons/react/solid'
+import { calculateMaxVoteScore } from '@models/proposal/calulateMaxVoteScore'
+import VotingPowerPct from './VotingPowerPct'
+import { BN } from '@coral-xyz/anchor'
+import Link from 'next/link'
+import useQueryContext from '@hooks/useQueryContext'
+import InlineNotification from '@components/InlineNotification'
+import { getTokenOwnerRecordAddress } from '@solana/spl-governance'
+import useVotePluginsClientStore from 'stores/useVotePluginsClientStore'
+
+interface Props {
+  className?: string
+}
+
+export default function LockedCommunityNFTRecordVotingPower(props: Props) {
+  const { fmtUrlWithCluster } = useQueryContext()
+  const [amount, setAmount] = useState(new BigNumber(0))
+  const {
+    councilMint,
+    mint,
+    realm,
+    ownTokenRecord,
+    realmTokenAccount,
+    symbol,
+    config,
+  } = useRealm()
+  const { proposal } = useProposal()
+  const [connected, wallet] = useWalletStore((s) => [s.connected, s.current])
+  const [tokenOwnerRecordPk, setTokenOwnerRecordPk] = useState('')
+  const [currentClient] = useVotePluginsClientStore((s) => [
+    s.state.currentRealmVotingClient,
+  ])
+  const [
+    loadingPositions,
+    votingPower,
+    amountLocked,
+    positions,
+    propagatePositions,
+  ] = useHeliumVsrStore((s) => [
+    s.state.isLoading,
+    s.state.votingPower,
+    s.state.amountLocked,
+    s.state.positions,
+    s.propagatePositions,
+  ])
+
+  useEffect(() => {
+    if (currentClient.heliumVsrVotingPositions.length !== positions.length) {
+      propagatePositions({ votingClient: currentClient })
+    }
+  }, [positions, currentClient, propagatePositions])
+
+  useEffect(() => {
+    if (mint && votingPower) {
+      setAmount(getMintDecimalAmount(mint, votingPower))
+    }
+  }, [mint, votingPower])
+
+  useEffect(() => {
```

Comment:
there is a query for exactly this in the queries folder, I suggest using it. I've seen this copy pasted around the codebase and it has race conditions (which you may have fixed, I didnt bother to look super close).

Reply:
useAddressQuery_TokenOwnerRecord, useAddressQuery_CommunityTokenOwner, useAddressQuery_CouncilTokenOwner